okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -798,7 +799,8 @@ private void proxyRequestResponse(Iterator<ModuleInstance> it,
       stream.handler(data -> {
         pc.trace("proxyRequestResponse request chunk '"
             + data.toString() + "'");
-        clientRequest.write(data);
+        stream.pause();
+        clientRequest.write(data, comp -> stream.resume());
         for (HttpClientRequest r : clientRequestList) {
           r.write(data);
```

Comment:
If r can handle less data per second than clientRequest the data in r's queue will grow until out of memory.

Reply:
More theoretical than practical. But I'll change and resume when all writes are done.